vdev/src/commands/test.rs

```diff
@@ -38,13 +38,14 @@ impl Cli {
         let runner = get_agent_test_runner(self.container)?;
 
         let mut args = vec!["--workspace".to_string()];
-        if let Some(extra_args) = &self.args {
-            args.extend(extra_args.clone());
 
-            if !(self.container || extra_args.contains(&"--features".to_string())) {
-                let features = platform::default_features();
-                args.extend(["--features".to_string(), features.to_string()]);
-            }
```

Comment:
The fact that this was in the `if let Some(...)` seems like a straight-up bug. I'm not sure why it also checked for `!container` previously. I couldn't find any reasoning for that looking back at the PR that introduced this https://github.com/vectordotdev/vector/pull/15833

Reply:
Agreed. I'm not sure why `!container` was there. Is there any harm in keeping it?

Reply:
Yeah, I agree that behavior seems unintended 👍🏻